pkg/flow/internal/controller/loader.go

```diff
@@ -392,6 +402,10 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
 		switch n := n.(type) {
 		case BlockNode:
 			err = l.evaluate(logger, parentScope, n)
+			if exp, ok := n.(*ExportConfigNode); ok {
+				name, val := exp.NameAndValue()
+				exports[name] = val
+			}
```

Comment:
Doesn't this suffer the same problem as before, where partial evaluates of a block node will cause other exports to get dropped? 

IMO we should probably be using a mechanism similar to `valueCache` (but for exports) here, where the cache gets cleared before a full evaluation, but updated in-place and re-exported any time a partial evaluation happens.

Reply:
I'm thinking how to recreate what you are worried on here to see how it works. I created a 2 export module loaded as `module.string` with content from a `local.file`. I then updated the value of 1 of the 2 exports and waited for `local.file` polling to reload the module. It seems to pick up the change and I can repeat the process numerous times.

Are we in a situation where since reload isn't technically what reloads the module contents in any way I can't recreate this yet? Or does this show it working as intended? Or maybe I'm missing the test case?

Reply:
Are you saying a situation where there is a failure in re-evaluating a module and the walk exits early? 

I think even using `valueCache` style behavior there will be issues if the reload fails. 

IMO until we define the failure mode of modules, failures during module parsing should be considered undefined behavior. For what its worth, I would prefer to cancel the whole load and reload from the last known good state for the WHOLE config in the case of any error.

Reply:
This should be able to reproduce the scenario:

```river 
discovery.kubernetes "nodes" {
  role = "node" 
}

discovery.kubernetes "pods" {
  role = "pod" 
}

export "nodes" {
  value = discovery.kubernetes.nodes.targets 
}

export "pods" {
  value = discovery.kubernetes.pods.targets 
}
```

On the initial load, both `export.nodes` and `export.pods` will be exposed to the parent module, but if either `discovery.kubernetes.nodes` or `discovery.kubernetes.pods` updates, it will trigger a partial reevaluation of the graph, and cause the other export to disappear.

Reply:
Apologies, misunderstood the situation you were describing, looking into this now.

Reply:
Re-used the `valueCache` since it generally mirrors the usage. Going to work on a unit test for this.